src/vaults/legacy/LegacyVaultAdapter.sol

```diff
@@ -55,7 +55,7 @@ contract LegacyVaultAdapter is AsyncVault, ILegacyVaultAdapter, IInvestmentManag
 
     // --- IInvestmentManager impl ---
     function escrow() public view returns (address) {
-        return address(_poolEscrowProvider.escrow(poolId));
+        return address(manager().globalEscrow());
```

Comment:
Note: Change seems correct to me given that legacy vaults use the escrow as target for `requestDeposit` and `requestRedeem`